### ISSUE TITLE
🐛 [cloud_functions] Can't call 2nd gen functions due to missing `httpsCallableFromURL` method

### ISSUE THREAD

**Issue by jhuleatt** [firebase/flutterfire#10541]
## Bug report

**Describe the bug**
2nd gen callable functions don't yet have stable URLs, so [an extra argument is needed to get the callable instance](https://firebase.google.com/docs/functions/beta/callable#call_the_function). Since FlutterFire doesn't support this, it isn't currently possible to call 2nd gen callable functions using FlutterFire.

### Expected behavior

FlutterFire offers a function that takes a callable function's URL as an argument, similar to iOS' `httpsCallable(URL)` overload, the JS SDK's `httpsCallableFromURL`, and Android's `getHttpsCallableFromUrl`.

## Additional context

1st gen functions have stable, guessable `cloudfunctions.net` URLs, so the client SDKs don't need to be told what a function's URL is. We're working on adding stable URLs to 2nd gen functions, but until they're available, URLs look like `https://<functionName>-<randomHash>-<region>.a.run.app/addMessage` and aren't guessable.

**Comment by jbaptisteroesch** [comment]
Hi,
I tested this solution for Gen2 functions. It appears that depending on the region, it may or may not work. I tested it on eu-west-3 and it did not work, but on eu-west-1 it did work. I'm not sure what the cause of this discrepancy is, but [this article led me to wonder if the problem might be related to the location](https://cloud.google.com/run/docs/mapping-custom-domains). It's possible that the location isn't related at all, but it's worth considering.
